### PR TITLE
ID-278 Indev integration

### PR DESCRIPTION
**Pull request by josealmeida**
https://nicedigital.atlassian.net/browse/ID-278

### CODE REVIEW

File: NICE.Identity.Authentication.Sdk/Extensions/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -36,6 +41,24 @@ public static AuthenticationBuilder AddAuthentication(this IServiceCollection se
...
+            {
+                var redisConfiguration =
+                    StackExchange.Redis.ConfigurationOptions.Parse(
+                        authConfiguration.RedisConfiguration.ConnectionString); 
+
+                services.AddSingleton<IRedisCacheClient, RedisCacheClient>();
+                services.AddSingleton<IRedisCacheConnectionPoolManager, RedisCacheConnectionPoolManager>();
+                services.AddSingleton<ISerializer, NewtonsoftSerializer>();
+
+                services.AddSingleton((provider) => provider.GetRequiredService<IRedisCacheClient>().GetDbFromConfiguration());
```
Comment (John-Williams) [inline NICE.Identity.Authentication.Sdk/Extensions/ServiceCollectionExtensions.cs:56]: isn't this provided by line 52?

File: NICE.Identity.Authentication.Sdk/TokenStore/RedisApiTokenStore.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        public RedisApiTokenStore(IRedisCacheClient cacheClient)
+        {
+            _cacheClient = cacheClient.GetDbFromConfiguration();
+        }
+#elif NET461
+        public RedisApiTokenStore(string redisConnectionString)
+        {
+            var serializer = new NewtonsoftSerializer(new JsonSerializerSettings
+            {
+                ReferenceLoopHandling = ReferenceLoopHandling.Ignore
```
Comment (John-Williams) [inline NICE.Identity.Authentication.Sdk/TokenStore/RedisApiTokenStore.cs:35]: why do you need this?

File: NICE.Identity.Authentication.Sdk/TokenStore/RedisApiTokenStore.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        {
+            var key = $"{KeyPrefix}{Guid.NewGuid().ToString()}";
+            await _cacheClient.AddAsync(key, token, TimeSpan.FromSeconds(token.ExpiresIn));
+            return key;
+        }
+
+        public void Dispose()
+        {
+#if NETFRAMEWORK
+            var endPoints = _cacheClient.Database.Multiplexer.GetEndPoints();
```
Comment (John-Williams) [inline NICE.Identity.Authentication.Sdk/TokenStore/RedisApiTokenStore.cs:80]: you might want to pass true here.
see:
https://github.com/StackExchange/StackExchange.Redis/blob/304970c06d6004970c9b190df940f32a9a68c087/StackExchange.Redis/StackExchange/Redis/ConnectionMultiplexer.cs#L481

File: NICE.Identity.Test/IntegrationTests/Authentication.Sdk/TokenStore/ApiTokenStoreTests.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+using NICE.Identity.Authentication.Sdk.TokenStore;
+using StackExchange.Redis.Extensions.Core.Abstractions;
+using StackExchange.Redis.Extensions.Core.Configuration;
+using StackExchange.Redis.Extensions.Core.Implementations;
+using StackExchange.Redis.Extensions.Newtonsoft;
+using Xunit;
+
+namespace NICE.Identity.Test.IntegrationTests.Authentication.Sdk.TokenStore
+{
+    [Trait("Category","Integration")]
```
Comment (John-Williams) [inline NICE.Identity.Test/IntegrationTests/Authentication.Sdk/TokenStore/ApiTokenStoreTests.cs:14]: nice

**Review by John-Williams — CHANGES_REQUESTED** [review]

File: NICE.Identity.Authentication.Sdk/Extensions/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (josealmeida) [reply NICE.Identity.Authentication.Sdk/Extensions/ServiceCollectionExtensions.cs:56]: That line sets up the default database from configuration using the GetDbFromConfiguration method.

File: NICE.Identity.Authentication.Sdk/TokenStore/RedisApiTokenStore.cs (hunk 2 above)
Comment (josealmeida) [reply NICE.Identity.Authentication.Sdk/TokenStore/RedisApiTokenStore.cs:35]: This is to ignore errors from reference loops when doing serialisation. It shouldn't be an issue due to the token class structure. It came from the code in RedisOwinSessionStore, but I suppose we can remove it from here.

**Review by John-Williams — APPROVED** [review]

**Review by John-Williams — APPROVED** [review]

**Pull request merged**